internal/work/dispatcher.go

```diff
@@ -297,99 +259,96 @@ func (d *Dispatcher) Transmit(
 	metadata map[string]string,
 	data []byte,
 ) (responseCode int, responseMetadata map[string]string, responseData []byte, responseError *dbus.Error) {
-	name, err := d.senderName(sender)
-	if err != nil {
+	ch := make(chan yggdrasil.Response)
+	d.Outbound <- struct {
+		Data yggdrasil.Data
+		Resp chan yggdrasil.Response
+	}{
+		Data: yggdrasil.Data{
+			Type:       yggdrasil.MessageTypeData,
+			MessageID:  messageID,
+			ResponseTo: responseTo,
+			Version:    1,
+			Sent:       time.Now(),
+			Directive:  addr,
+			Metadata:   metadata,
+			Content:    data,
+		},
+		Resp: ch,
+	}
+
+	select {
+	case resp := <-ch:
+		responseCode = resp.Code
+		responseMetadata = resp.Metadata
+		responseData = resp.Data
+	case <-time.After(1 * time.Second):
```

Comment:
Why is timeout only one second? I think that this is too low value. Could it be configured somehow? Could it be e.g. part of MQTT message sent to worker? If the `timeout` would not be part of MQTT message, then some default value would be used.

Reply:
With the separation of `Transmit` and `Request`, this timeout now applies only
to the configured data transport's implementation of `Tx`. This timeout is
waiting for a response to the underlying transport `Tx` call. In practice, this
happens instantaneously with the MQTT transport because the MQTT transport does
not support responses. The HTTP transport does respond to `Transmit` requests
with the response of the underlying HTTP request. This is an unfortunate side
effect of supporting both MQTT and HTTP as data transports.

I am not sure whether it is worth exposing this as a configurable value. Maybe
we could re-use the existing `config.HTTPTimeout` configuration value?